pkg/mod/mod.go

```diff
@@ -0,0 +1,40 @@
+package mod
+
+import (
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type goModule struct {
+	Path     string     // module path
+	Version  string     // module version
+	Versions []string   // available module versions (with -versions)
+	Replace  *goModule  // replaced by this module
+	Time     *time.Time // time version was created
+	Update   *goModule  // available update, if any (with -u)
+	Main     bool       // is this the main module?
+	Indirect bool       // is this module only an indirect dependency of main module?
+	Dir      string     // directory holding files for this module, if any
+	GoMod    string     // path to go.mod file for this module, if any
+}
+
+type goModules []*goModule
+
+var GoMods goModules
```

Comment:
you should be getting a lint error from this, globals are forbidden, they serve almost no purpose and should likewise almost never be used! I don't see a valid reason to allow this one.